keystore/src/connection/platform/generic/mod.rs

```diff
@@ -119,7 +121,17 @@ impl SqlCipherConnection {
         Ok(conn)
     }
 
-    fn init_with_key_in_memory(_path: &str, key: &str) -> CryptoKeystoreResult<Self> {
+    fn set_key(conn: &mut rusqlite::Connection, key: &DatabaseKey) -> CryptoKeystoreResult<()> {
+        // Make sqlite use raw key data, without key derivation. Also make sure to zeroize
+        // the string containing the key after the call.
+        let mut key = format!("x'{}'", hex::encode(key));
```

Comment:
Looking at [the docs](https://www.zetetic.net/sqlcipher/sqlcipher-api/#key), I think the hex key may need to be uppercase. 
```suggestion
        let mut key = format!("x'{}'", hex::encode_upper(key));
```

Reply:
Lowercase also works fine, I tested it.